### PR TITLE
Test perl 5.26 and avoid JSON problems in test suite with it

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

**Comment by okurz** [comment]
LGTM

**Comment by coolo** [comment]
left side it fails - it also takes much longer before travis finished the now 8 jobs. So I think we need to decide which perl version we want - or do a wild mix of perl versions in the matrix.

@mudler @foursixnine - what do you think?

File: cpanfile [hunk 1]
```diff
@@ -36,8 +36,7 @@ requires 'IO::Handle';
 requires 'IO::Socket::SSL';
 requires 'IPC::Cmd';
 requires 'IPC::Run';
-requires 'JSON';
```
Comment (mudler) [inline cpanfile]: (minor) this is still used during tests

**Comment by mudler** [comment]
@coolo : +1 for switching to Cpanel::JSON::XS.

For perl versions: targeting more of them it's always a good idea imho (expecially new ones) but the side effect will be a very slow travis build (and as tests are still a bit flacky that means: 1 failure = travis test failure = lot of restarts).

If we have to pick one, i'd go with the latest ones - but in a ideal world, to not break compatibility we should test also old versions as well.

**Comment by coolo** [comment]
well, openSUSE and SUSE run production on  5.18 - and for TW we have the test suite in the package, so we notice early if there is a problem. So I'd opt to stay with 5.18 here.

**Comment by foursixnine** [comment]
I'd opt for 5.18 too.

**Comment by okurz** [comment]
so it's either faster feedback for each PR before it reaches OBS but longer waiting time for each PR or wait for OBS and accept broken state in github

**Comment by coolo** [comment]
yeah - the typical judgement: optimize for success or optimize for failure.

**Comment by coolo** [comment]
I find it suprising though that we have increased coverage with this :)

So perhaps we leave it as is for a while

**Comment by okurz** [comment]
I don't find it surprising that this would increase coverage. Run the flaky coverage test multiple times and take the highest coverage of all the runs, done :)

**Comment by coolo** [comment]
true, but AMQP.pm increased 0.64?

**Comment by okurz** [comment]
yeah ok, not this one ;) So let's agree it *is* surprising

**Pull request merged**